### PR TITLE
TST: Remove crackfortran.nameargspattern time test that failed randomly

### PR DESCRIPTION
**Pull request by molsonkiko**
Address @seberg 's comment on PR #23460 that the median doubling test failed too much even though the regex is good.
I removed that test to address the problem.
Also made the threshold for rejecting a regex as too slow much more lenient.
200ms should be enough time even for a bad CPU on a bad day, while a bad regex should fail with near certainty even on super fast CPUs.

I ran all the tests like 20 times on my own machine and it never failed once. I then subbed in the *old* nameargspattern (`re.compile(r'\s*(?P<name>\b[\w$]+\b)\s*(@\(@\s*(?P<args>[\w\s,]*)\s*@\)@|)\s*((result(\s*@\(@\s*(?P<result>\b[\w$]+\b)\s*@\)@|))|(bind\s*@\(@\s*(?P<bind>.*)\s*@\)@))*\s*\Z', re.I)`) and it failed every time I tested it (which was about 5 times because I'm not *that* patient).

Bottom line: the tests will now run faster and bad regexes should fail in less than 10 seconds.

### CODE REVIEW

**Comment by seberg** [comment]
Thanks for quickly following up.  Not sure how bad it was, it was just one failure.  But let's try this.
